hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/ReflectionUtils.java

```diff
@@ -61,6 +101,72 @@ private ReflectionUtils() {
         return (T) field.get(null);
     }
 
+    /**
+     * Return a set-method for a class and a property. The setter must start
+     * with "set", must be public, non-static, must return void and take one
+     * argument of type {@code type}.
+     *
+     * @param clazz The containing class
+     * @param propertyName Name of the property
+     * @param type The type of the property
+     *
+     * @return The found setter or null if one matching the criteria doesn't exist
+     */
+    @Nullable
+    public static Method findPropertySetter(
+            @Nonnull Class<?> clazz,
+            @Nonnull String propertyName,
+            @Nonnull Class<?> type
+    ) {
+        String setterName = "set" + toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
+
+        Method method;
+        try {
+            method = clazz.getMethod(setterName, type);
+        } catch (NoSuchMethodException e) {
+            return null;
+        }
+
+        if (!Modifier.isPublic(method.getModifiers())) {
+            return null;
+        }
+
+        if (Modifier.isStatic(method.getModifiers())) {
+            return null;
+        }
+
+        Class<?> returnType = method.getReturnType();
+        if (returnType != void.class && returnType != Void.class) {
+            return null;
```

Comment:
AFAIU this rules out builder-style setters. Do we really need to reject such setters?

Reply:
fixed